examples/demo-app/src/cloud-providers/aws/aws-provider.js

```diff
@@ -0,0 +1,291 @@
+        authWindow.close();
+      }
+
+      window.removeEventListener('message', handleLogin);
+
+      if (e.data.success) {
+        onCloudLoginSuccess();
+      }
+    };
+    window.addEventListener('message', handleLogin);
+  }
+
+  /**
+   * Required!
+   *
+   * @returns {Array<Viz>}
+   */
+  async listMaps() {
+    let s3PrivateFiles = await this._getMapListFromStorage('private');
+    let s3PublicFiles = await this._getMapListFromStorage('public');
+
+    return [...s3PublicFiles, ...s3PrivateFiles];
+  }
+
+  async _getMapListFromStorage(level) {
+    let mapList = [];
+    let capitalizedLevel = level.charAt(0).toUpperCase() + level.slice(1);
+    await Storage.list('', {level})
+      .then((result) => {
+        mapList = AwsProvider._updateFileList(result, level);
+      })
+      .catch((e) => {
+        this._handleError(`${capitalizedLevel} map load failed`, e);
+      });
+    return mapList;
+  }
+
+  static async _updateFileList(fileList, level) {
+    let updatedFileList = [];
+
+    for (const file of fileList) {
+      if (!file.key.endsWith('.json')) {
+        continue;
+      }
+      const title = file.key.split('.')[0];
+      const thumbnailKey = title + '.png';
+      const thumbnailURL = await Storage.get(thumbnailKey, {
+        level: level,
+        download: false
+      });
+
+      const description = await Storage.get(thumbnailKey, {
+        level: level,
+        download: true
+      }).then((thumbnail) =>
+        thumbnail.Metadata.desc ? atob(thumbnail.Metadata.desc) : 'No description available.'
+      );
+      const mapIsPrivate = level === 'private';
+
+      updatedFileList.push({
+        id: file.key,
+        title: title,
+        description: description,
+        privateMap: mapIsPrivate,
+        thumbnail: thumbnailURL,
+        lastModification: new Date(Date.parse(file.lastModified)),
+        loadParams: {
+          mapId: file.key,
+          privateMap: mapIsPrivate,
+          level: level
+        }
+      });
+    }
+    return updatedFileList;
+  }
+
+  /**
+   * Required!
+   *
+   * @returns {MapResponse}
+   */
+  async downloadMap(loadParams) {
+    const {level, mapId, identityId} = loadParams;
+    let mapURL =
+      level === 'private'
+        ? await Storage.get(mapId, {level})
+        : await Storage.get(mapId, {level, identityId});
+
+    const mapData = await fetch(mapURL).then((response) => response.json());
+    this._loadParam = loadParams;
+
+    return {
+      map: mapData,
+      format: 'keplergl'
+    };
+  }
+
+  async uploadMap({mapData, options = {}}) {
+    const {isPublic, overwrite} = options;
+    const {map, thumbnail} = mapData;
+    const {title, description} = map && map.info;
+    const name = title;
+    // Since we share through a map link, this could be private as well
+    const level = isPublic ? 'protected' : 'private';
+    let mapId = '';
+    let identityId = '';
+
+    await Auth.currentUserInfo().then((userInfo) => {
```

Comment:
Same comment as in line https://github.com/deepatlas/kepler.gl/pull/1/files#diff-9cc6944f72643588ae105aaffcf246beR81